operator/internal/controller/vectorized/cluster_controller_test.go

```diff
@@ -0,0 +1,386 @@
+					Port: 33145,
+				},
+				PandaproxyAPI: []vectorizedv1alpha1.PandaproxyAPI{
+					{
+						Name: "proxy-internal",
+						Port: 8082,
+						External: vectorizedv1alpha1.PandaproxyExternalConnectivityConfig{
+							ExternalConnectivityConfig: vectorizedv1alpha1.ExternalConnectivityConfig{
+								Enabled: false,
+							},
+						},
+					},
+					{
+						Name: "proxy-external",
+						External: vectorizedv1alpha1.PandaproxyExternalConnectivityConfig{
+							ExternalConnectivityConfig: vectorizedv1alpha1.ExternalConnectivityConfig{
+								Enabled:   true,
+								Subdomain: "test.example.com",
+							},
+							Ingress: &vectorizedv1alpha1.IngressConfig{
+								Enabled:  ptr.To(true),
+								Endpoint: "proxy",
+								Annotations: map[string]string{
+									"nginx.ingress.kubernetes.io/ssl-passthrough": "true",
+								},
+							},
+						},
+					},
+				},
+				SchemaRegistry: &vectorizedv1alpha1.SchemaRegistryAPI{
+					Port: 8081,
+				},
+			},
+			Resources: vectorizedv1alpha1.RedpandaResourceRequirements{
+				ResourceRequirements: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("1"),
+						corev1.ResourceMemory: resource.MustParse("2Gi"),
+					},
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("1"),
+						corev1.ResourceMemory: resource.MustParse("2Gi"),
+					},
+				},
+			},
+		},
+	}
+}
+
+func minimalClusterNormalizer(ctx context.Context, testScheme *runtime.Scheme, cluster *vectorizedv1alpha1.Cluster, t *testing.T, ctl *kube.Ctl) []client.Object {
+	var allClusterResources []client.Object
+	var lists []client.ObjectList
+	for gvk := range testScheme.AllKnownTypes() {
+		if strings.HasSuffix(gvk.Kind, "List") {
+			obj, err := testScheme.New(gvk)
+			require.NoError(t, err)
+			if l, ok := obj.(client.ObjectList); ok {
+				lists = append(lists, l)
+			}
+		}
+	}
+
+	clusterSelector := labels.ForCluster(cluster).AsClientSelector()
+	for _, l := range lists {
+		err := ctl.List(ctx, l)
+		if errors.Is(err, &meta.NoKindMatchError{}) {
+			continue
+		}
+		require.NoError(t, err)
+
+		objs, err := meta.ExtractList(l)
+		require.NoError(t, err)
+
+		for _, obj := range objs {
+			o := obj.(client.Object)
+
+			if !clusterSelector.Matches(k8slabels.Set(o.GetLabels())) {
+				continue
+			}
+
+			allClusterResources = append(allClusterResources, o)
+			t.Logf("%T: %s/%s", o, o.GetNamespace(), o.GetName())
+		}
+	}
+
+	for _, rs := range allClusterResources {
+		rs.SetResourceVersion("1")
+		rs.SetUID("1234")
+		rs.SetCreationTimestamp(metav1.Unix(0, 0))
+		rs.SetManagedFields(nil)
+		if ownerRefs := rs.GetOwnerReferences(); len(ownerRefs) > 0 {
+			for i := range ownerRefs {
+				ownerRefs[i].UID = "normalized-uid-1234"
+			}
+			rs.SetOwnerReferences(ownerRefs)
+		}
+
+		if annotations := rs.GetAnnotations(); annotations != nil {
+			annotations["redpanda.com/last-applied"] = "applied"
```

Comment:
I would expect this to be deterministic actually 🤔 

I'd still ask for an obviously fake value. `<removed>` is always a good option.